pkg/cloudprovider/azure.go

```diff
@@ -63,17 +64,27 @@ func (a *Azure) initCredentials() error {
 		return err
 	}
 
-	a.vmClient = compute.NewVirtualMachinesClient(subscriptionID)
+	// Pick the Azure "Environment", which is just a named set of API endpoints.
+	if a.cfg.APIOverride != "" {
+		a.env, err = azure.EnvironmentFromURL(a.cfg.APIOverride)
+	} else {
+		a.env, err = azure.EnvironmentFromName(a.cfg.AzureEnvironment)
+	}
```

Comment:
So there always needs to be either an `APIOverride` or `AzureEnvironment` ?

Reply:
Kind of, there is a default Azure environment specified. But Azure makes liberal use of the "environment" -- you even need to change it if you're using the Azure Germany cloud.